### PR TITLE
Remove deprecated webpacker spelling

### LINKED ISSUE #398: Drop backward compatibility for Webpacker spelling in v8.0.0
Shakapacker v7 changed the spelling of Webpacker to Shakapacker in the entire project but still provided backward compatibility for Webpacker spelling.

We should drop this backward compatibility in the next major release.

### PR DESCRIPTION
**Pull request by tomdracz**
### Summary

Closes #398

Drops backwards compatibility with Webpacker spelling, constants, ENV variables etc that have been deprecated in v7.

### Pull Request checklist

- [x] Add/update test to cover these changes
- [x] Update documentation
- [x] Update CHANGELOG file

### CODE REVIEW

**Comment by tomdracz** [comment]
@justin808 I think some remaining failures on this one are due to react-on-rails being used in the dummy app and that one still referencing `Webpacker` constants

https://github.com/shakacode/shakapacker/actions/runs/8177448565/job/22359120922#step:6:206

`Webpacker` heavily used in https://github.com/shakacode/react_on_rails/blob/master/lib/react_on_rails/webpacker_utils.rb

Any plans to drop that/support both Webpacker/Shakapacker?

File: spec/dummy/config/application.rb [hunk 1]
```diff
@@ -19,6 +19,9 @@
 # you've limited to :test, :development, or :production.
 Bundler.require(*Rails.groups)
 
+# Temp patch for dummy specs
```
Comment (tomdracz) [inline spec/dummy/config/application.rb:22]: As per https://github.com/shakacode/shakapacker/pull/429#issuecomment-1981813624 - this is temp patch for dummy specs that are using `react-on-rails` as that still expects `Webpacker` constant to be present. Can be removed once we address in `react-on-rails`

**Comment by ahangarha** [comment]
This PR is more like removing backward compatibility rather than removing deprecation messages.

Am I right?

**Comment by tomdracz** [comment]
@ahangarha correct! Removes all backwards compatibility layer ends deprecated changes it was trying to keep compatible in the deprecation period!

**Review by G-Rath — APPROVED** [review]
woot woot! :shipit:

**Review by justin808 — APPROVED** [review]
Looks GREAT. But I skimmed the diffs. 

How confident are @tomdracz and @G-Rath and @ahangarha ? If you all are, I'll merge!

**Comment by tomdracz** [comment]
Pretty confident @justin808 Test suite should give us enough confidence and I ran some sense checks with no issues. @ahangarha initial work in adding this was nicely commented and named so it was easy enough to find all instances to remove.

Should be good to go!

**Comment by G-Rath** [comment]
+ we're landing this on `next`, so we'll have some time to find anything we've missed as we're landing other breaking changes :)

**Comment by tomdracz** [comment]
Merging as I know you have some PRs in flight @G-Rath that will probably warrant from rebase after this!

**Comment by ahangarha** [comment]
I am happy that my approach in adding backward compatibility made it easy to remove it now 👍🏾

**Pull request merged**